// phonepesdk/artifacts/constants/PhonePeSDKExternalConstants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ExternalConstants;
(function (ExternalConstants) {
    let Permission;
    (function (Permission) {
        Permission["READ_SMS"] = "READ_SMS";
        Permission["LOCATION"] = "LOCATION";
        Permission["CAMERA"] = "CAMERA";
        Permission["WRITE_EXTERNAL_STORAGE"] = "WRITE_EXTERNAL_STORAGE";
        Permission["READ_EXTERNAL_STORAGE"] = "READ_EXTERNAL_STORAGE";
    })(Permission = ExternalConstants.Permission || (ExternalConstants.Permission = {}));
    let OrientationMode;
    (function (OrientationMode) {
        OrientationMode["LANDSCAPE"] = "LANDSCAPE";
        OrientationMode["PORTRAIT"] = "PORTRAIT";
        OrientationMode["DEVICE"] = "DEVICE";
    })(OrientationMode = ExternalConstants.OrientationMode || (ExternalConstants.OrientationMode = {}));
    let Species;
    (function (Species) {
        Species["web"] = "web";
        Species["native"] = "native";
    })(Species = ExternalConstants.Species || (ExternalConstants.Species = {}));
    let OS;
    (function (OS) {
        OS["ios"] = "ios";
        OS["android"] = "android";
    })(OS = ExternalConstants.OS || (ExternalConstants.OS = {}));
    let OSUserAgent;
    (function (OSUserAgent) {
        OSUserAgent["ios"] = "phonepe-ios";
        OSUserAgent["android"] = "phonepe-android";
    })(OSUserAgent = ExternalConstants.OSUserAgent || (ExternalConstants.OSUserAgent = {}));
    let UserDetail;
    (function (UserDetail) {
        UserDetail["email"] = "email";
        UserDetail["name"] = "name";
        UserDetail["phoneNumber"] = "phoneNumber";
        UserDetail["isEmailVerified"] = "isEmailVerified";
    })(UserDetail = ExternalConstants.UserDetail || (ExternalConstants.UserDetail = {}));
    let BlueToothListenerEvents;
    (function (BlueToothListenerEvents) {
        BlueToothListenerEvents["BleManagerDidUpdateState"] = "BleManagerDidUpdateState";
        BlueToothListenerEvents["BleManagerStopScan"] = "BleManagerStopScan";
        BlueToothListenerEvents["BleManagerDiscoverPeripheral"] = "BleManagerDiscoverPeripheral";
        BlueToothListenerEvents["BleManagerDidUpdateValueForCharacteristic"] = "BleManagerDidUpdateValueForCharacteristic";
        BlueToothListenerEvents["BleManagerConnectPeripheral"] = "BleManagerConnectPeripheral";
        BlueToothListenerEvents["BleManagerDisconnectPeripheral"] = "BleManagerDisconnectPeripheral";
    })(BlueToothListenerEvents = ExternalConstants.BlueToothListenerEvents || (ExternalConstants.BlueToothListenerEvents = {}));
    ExternalConstants.version = '1.0.4';
})(ExternalConstants = exports.ExternalConstants || (exports.ExternalConstants = {}));
