// phonepesdk/artifacts/modules/font/DynamicFonts.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const react_native_1 = require("react-native");
class DynamicFonts {
    static loadFont(name, data, type, forceLoad) {
        if (!forceLoad && this.loadedFonts[name])
            return Promise.resolve(this.loadedFonts[name]);
        if (!name)
            throw new Error('Name is a required argument');
        if (!data)
            throw new Error('Data is a required argument');
        return new Promise((resolve, reject) => {
            react_native_1.NativeModules.DynamicFonts.loadFont({
                name: name,
                data: data,
                type: type
            }, ((err, givenName) => {
                if (err) {
                    reject(err);
                    return;
                }
                this.loadedFonts[name] = givenName;
                resolve(givenName);
            }));
        });
    }
    static loadFonts(_fontList, forceLoad) {
        let fontList = _fontList;
        if (!fontList)
            return Promise.resolve([]);
        if (!(fontList instanceof Array))
            fontList = [fontList];
        return Promise.all(fontList.filter((font) => font).map((font) => this.loadFont(font.name, font.data, font.type, forceLoad)));
    }
}
DynamicFonts.loadedFonts = {};
exports.DynamicFonts = DynamicFonts;
