// phonepesdk/artifacts/modules/contact/model.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ContactNameSpace;
(function (ContactNameSpace) {
    let CONTACT_PICKER_MODE;
    (function (CONTACT_PICKER_MODE) {
        CONTACT_PICKER_MODE[CONTACT_PICKER_MODE["MODE_SINGLE_SELECT"] = 2] = "MODE_SINGLE_SELECT";
        CONTACT_PICKER_MODE[CONTACT_PICKER_MODE["MODE_MULTI_SELECT"] = 1] = "MODE_MULTI_SELECT";
    })(CONTACT_PICKER_MODE = ContactNameSpace.CONTACT_PICKER_MODE || (ContactNameSpace.CONTACT_PICKER_MODE = {}));
    let CONTACT_PICKER_TYPE;
    (function (CONTACT_PICKER_TYPE) {
        CONTACT_PICKER_TYPE[CONTACT_PICKER_TYPE["TYPE_CONTACT_PHONE"] = 0] = "TYPE_CONTACT_PHONE";
        CONTACT_PICKER_TYPE[CONTACT_PICKER_TYPE["TYPE_CONTACT_VPA"] = 1] = "TYPE_CONTACT_VPA";
        CONTACT_PICKER_TYPE[CONTACT_PICKER_TYPE["TYPE_CONTACT_ACCOUNT"] = 2] = "TYPE_CONTACT_ACCOUNT";
        CONTACT_PICKER_TYPE[CONTACT_PICKER_TYPE["TYPE_MIXED"] = 3] = "TYPE_MIXED";
    })(CONTACT_PICKER_TYPE = ContactNameSpace.CONTACT_PICKER_TYPE || (ContactNameSpace.CONTACT_PICKER_TYPE = {}));
})(ContactNameSpace = exports.ContactNameSpace || (exports.ContactNameSpace = {}));
